### PR TITLE
[16.0][ADD] report_qweb_decimal_precision

### PR DESCRIPTION
**Pull request by yostashiro**
This module allows administrators to define the decimal precision of float fields for QWeb report presentation.

Functionality wise, this basically replaces https://github.com/OCA/reporting-engine/tree/16.0/report_qweb_decimal_place.

~Tests need to be added.~

Configuration:
image

Result:
image

@qrtl

### CODE REVIEW

File: report_qweb_decimal_precision/models/ir_qweb_fields.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if "precision" not in options and "decimal_precision" not in options:
+            dp_qweb_recs = self.env["decimal.precision.qweb"].search(
+                [("res_model_name", "=", record._name), ("field_name", "=", field_name)]
+            )
+            score = 0
+            precision_rec = False
+            for dp_qweb_rec in dp_qweb_recs:
+                score_rec = dp_qweb_rec._get_score(record)
+                if score_rec > score:
+                    precision_rec = dp_qweb_rec
```
Comment (len-foss) [inline report_qweb_decimal_precision/models/ir_qweb_fields.py]:
```suggestion
            precision_rec = max(dp_qweb_recs, default=None, key=lambda r: r._get_score(record))
```

**Review by len-foss — APPROVED** [review]
Code LGTM

Feature is a very nice improvement :-)

File: report_qweb_decimal_precision/models/ir_qweb_fields.py (hunk 1 above)
Comment (yostashiro) [reply report_qweb_decimal_precision/models/ir_qweb_fields.py]: @len-foss Thanks for the suggestion! Updated the code.

**Review by AungKoKoLin1997 — APPROVED** [review]
LGTM.

**Comment by liuhehe1995** [comment]
It worked as expected.
Functional tests were performed.(purchase.order.line/sale.order.line/account.move.line)

**Review by liuhehe1995 — APPROVED** [review]
LGTM.

**Comment by HviorForgeFlow** [comment]
Hello @yostashiro and @AungKoKoLin1997 I can see that both module that you are proposing share similar functionalities. Should I merge this or should we go for https://github.com/OCA/reporting-engine/pull/857?

I can see that report_qwebt_field_converter is kind of more generic feature.

Feedback welcome 

CC @OCA/reporting-engine-maintainers

**Comment by yostashiro** [comment]
@HviorForgeFlow Indeed, #857 intends to supersede this PR if it works out nicely. We just haven't gotten around to do a review on that PR, which we will follow up in the coming days.

**Comment by HviorForgeFlow** [comment]
Ok thanks for your quick feedback, maybe this PR can be closed then.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Pending superseed https://github.com/OCA/reporting-engine/pull/857

**Pull request closed without merging**